docs/pages/api/access-plugin.mdx

```diff
@@ -0,0 +1,738 @@
+|`github.com/gravitational/teleport/api/types`|Types used in the Auth Service API, e.g., Access Requests.|
+|`github.com/gravitational/trace`|Presenting errors with more useful detail than the standard library provides.|
+|`google.golang.org/api/option`|Settings for configuring Google API clients.|
+|`google.golang.org/api/sheets/v4`|The Google Sheets API client library, aliased as `sheets` in our program.|
+|`google.golang.org/grpc`|The gRPC client and server library.|
+
+### Make some initial declarations
+
+After the `import` block, add the following declarations:
+
+```go
+const (
+	proxyAddr     string = ""
+	initTimeout          = time.Duration(30) * time.Second
+	spreadSheetID string = ""
+)
+
+var requestStates = map[types.RequestState]string{
+	types.RequestState_APPROVED: "APPROVED",
+	types.RequestState_DENIED:   "DENIED",
+	types.RequestState_PENDING:  "PENDING",
+	types.RequestState_NONE:     "NONE",
+}
+
+type googleSheetsPlugin struct {
+	sheetsClient   *sheets.SpreadsheetsService
+	teleportClient *client.Client
+}
+```
+
+`proxyAddr` indicates the address and port of your Teleport Proxy Service or
+Teleport Enterprise Cloud tenant. `initTimeout` is the maximum time we will wait
+to connect to the Auth Service, in this case, 30 seconds.
+
+Assign `spreadSheetID` to the ID of the spreadsheet you created earlier. To find
+the spreadsheet ID, visit your spreadsheet in Google Drive. The ID will be in
+the URL path segment called `SPREADSHEET_ID` below:
+
+```text
+https://docs.google.com/spreadsheets/d/SPREADHSEET_ID/edit#gid=0
+```
+
+Access Requests have one of four states: approved, denied, pending, and none.
+Later in the guide, we will need a way to map these states to strings that we
+can write to our Google Sheets spreadsheet. We obtain the request states from
+Teleport's `types` library and map them to strings in the `requestStates` map.
+
+`googleSheetsPlugin` is a struct type that contains the API clients for the
+Teleport Auth Service and Google Sheets. We will explain later why we use a
+struct to organize these two clients, instead of, say, two separate variables.
+
+### Prepare row data
+
+Whether creating a new row of the spreadsheet or updating an existing one, we
+need a way to extract data from an Access Reqeust in order to provide it to
+Google Sheets. Below the `googleSheetsPlugin` type declaration, add the
+following function:
+
+```go
+func (g *googleSheetsPlugin) makeRowData(ar types.AccessRequest) *sheets.RowData {
+
+	t := ar.GetCreationTime().String()
+	u := ar.GetUser()
+	s := strings.Join(ar.GetRoles(), ",")
+	i := ar.GetName()
+	a, ok := requestStates[ar.GetState()]
+	l := fmt.Sprintf(
+		"=HYPERLINK(\"%v\", \"%v\")",
+		"https://"+proxyAddr+"/web/requests/"+ar.GetName(),
+		"View Access Request",
+	)
+
+	// Could not find a state, but this is still a valid Access Request
+	if !ok {
```

Comment:
Move this right after the `a, ok :=` assignment. Having code in between makes this harder to follow.